link/src/main/java/com/stripe/android/link/account/LinkAccountManager.kt

```diff
@@ -32,57 +29,33 @@ internal class LinkAccountManager @Inject constructor(
     private val linkEventsReporter: LinkEventsReporter
 ) {
     private val _linkAccount = MutableStateFlow<LinkAccount?>(null)
-    var linkAccount: StateFlow<LinkAccount?> = _linkAccount
+    val linkAccount: StateFlow<LinkAccount?> = _linkAccount
+
+    @Volatile
+    @VisibleForTesting
+    var authSessionCookie: String? = null
 
     /**
      * The publishable key for the signed in Link account.
      */
+    @Volatile
+    @VisibleForTesting
     var consumerPublishableKey: String? = null
 
-    val accountStatus = linkAccount.transform { value ->
-        emit(
-            // If we already fetched an account, return its status
-            value?.accountStatus
-                ?: (
-                    // If consumer has previously logged in, fetch their account
-                    cookieStore.getAuthSessionCookie()?.let {
-                        lookupConsumer(null).map {
-                            it?.accountStatus
-                        }.getOrElse {
-                            AccountStatus.Error
-                        }
-                    }
-                        // If the user recently signed up on this device, use their email
-                        ?: cookieStore.getNewUserEmail()?.let {
-                            lookupConsumer(it).map {
-                                it?.accountStatus
-                            }.getOrElse {
-                                AccountStatus.Error
-                            }
-                        }
-                        // If a customer email was passed in, lookup the account,
-                        // unless the user has logged out of this account
-                        ?: config.customerEmail?.let { customerEmail ->
-                            if (hasUserLoggedOut(customerEmail)) {
-                                AccountStatus.SignedOut
-                            } else {
-                                lookupConsumer(customerEmail).map {
-                                    it?.accountStatus
-                                }.getOrElse {
-                                    AccountStatus.Error
-                                }
-                            }
-                        } ?: AccountStatus.SignedOut
-                    )
-        )
+    val accountStatus = linkAccount.map { value ->
+        // If we already fetched an account, return its status
+        value?.accountStatus
+            // If a customer email was passed in, lookup the account,
+            // unless the user has logged out of this account
+            ?: config.customerEmail?.let { customerEmail ->
```

Comment:
Where did the `If the user recently signed up on this device, use their email` part go?

In fact, `CookieStore.getNewUserEmail()` is unused now. Is this an oversight or can we get rid of it?

Reply:
I left new user email so that I could use it in a follow up, but it won't be used the same way it was before.

The plan is to have logic which determines the right email to use (the one the merchant passed in vs the one they signed up with, vs the last one they used to auth).